_posts/2019-03-14-2019-kafka-connector.md

```diff
@@ -0,0 +1,194 @@
+
+Here is a diagram which shows how the component works:
+
+![Form Options](/images/kafka-connector/kafka-connector-detailed-overview.png)
+
+## Pre-requisites
+
+First and most important thing is [running OpenFaaS instance](https://docs.openfaas.com/deployment/kubernetes/) with default configuration on Kubernetes, for our case [minikube deployment](https://kubernetes.io/docs/setup/minikube/) is more than enough.
+
+The [faas-cli](https://docs.openfaas.com/cli/install/) binary along with populated `OPENFAAS_URL` environmental variable pointing at your gateway.
+
+## Clone the Kafka Connector
+
+Pull the `kafka-connector` code from github:
+
+```bash
+$ git clone https://github.com/openfaas-incubator/kafka-connector
+```
+
+Inside `kafka-connector` folder navigate to `yaml/kubernetes`:
+
+```bash
+$ cd kafka-connector/yaml/kubernetes
+```
+
+Here we can see the Kafka Zookeeper and Broker which we will be using to send payloads to our functions.
+
+## Deploy the components with configuration
+
+In order to deploy the Zookeeper and Broker apply the kubernetes files while inside the kubernetes folder from above:
+
+```bash
+$ kubectl apply -f kafka-broker-dep.yml,kafka-broker-svc.yml,zookeeper-dep.yaml,zookeeper-svc.yaml
+```
+
+It is important to note that those files deploy the Zookeeper and Broker in the same namespace as the OpenFaaS Gateway.
+
+We will use charts to deploy the Kafka Connector. If you don't have the OpenFaaS charts repository, then add it with this command:
+
+```bash
+$ helm repo add openfaas https://openfaas.github.io/faas-netes/
+```
+
+Install Kafka Connector with default values:
+
+```bash
+$ helm upgrade kafka-connector openfaas/kafka-connector \
+    --install \
+    --namespace openfaas \
+    --set topics="faas-request\," \
+    --set broker_host="kafka" \
+    --set print_response_body="true"
+```
+
+Here it is important to note that our Broker resides in service called `kafka` in the same namespace as our Gateway and Kafka Connector service. If you have Broker service in another namespace, again called `kafka`, just append the namespace like so `kafka.<your_namespace>` to the `broker_host` environmental variable so the Connector can discover it. The default kafka port `9092` is appended to the service name by default.
+
+Also if you would like to subscribe the connector and your functions to more than one topic, list them in the `topics` variable by separating them with comma and re-run the command above.
+
+Congratulations now you have the whole stack deployed:
+* Zookeeper
+* Broker Host with Producer
+* Kafka Connector
+
+Before we proceed, make sure you have all the components running:
+
+```bash
+$ kubectl get pods -n openfaas | grep -E 'kafka|zookeeper|producer'
```

Comment:
why do we grep for producer?

Reply:
Alex requested in the upper comments to be more specific with the parts